libraries/c_sdk/standard/mqtt/test/unit/iot_tests_mqtt_receive.c

```diff
@@ -1585,15 +1645,15 @@ TEST( MQTT_Unit_Receive, UnsubackInvalid )
         TEST_ASSERT_EQUAL_INT( true, _processBuffer( &unsubscribe,
                                                      pUnsuback,
                                                      unsubackSize,
-                                                     IOT_MQTT_BAD_RESPONSE ) );
+                                                     IOT_MQTT_STATUS_PENDING ) );
 
         /* Network close should have been called for invalid packet. */
         TEST_ASSERT_EQUAL_INT( true, _networkCloseCalled );
         TEST_ASSERT_EQUAL_INT( true, _disconnectCallbackCalled );
         _networkCloseCalled = false;
         _disconnectCallbackCalled = false;
     }
-
+    IotListDouble_RemoveHead( &( _pMqttConnection->pendingResponse ) );
```

Comment:
and this one?

Reply:
Provided the comment for this addition in the code.